xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetabledate.js

```diff
@@ -0,0 +1,93 @@
+    'daterangepicker': ['jquery', 'bootstrap', 'moment']
+  }
+});
+
+require(['jquery', 'moment', 'daterangepicker'],
+  function($, moment) {
+    var dateFormat = moment().toMomentFormatString('$escapetool.javascript($xwiki.getXWikiPreference("dateformat", "yyyy/MM/dd HH:mm"))');
+
+    var bindInputs = function(bind) {
+      $(bind).find('input[data-type="date"]').each(function(i, element) {
+        var input = $(element);
+        var hidden = input.prev('input[type="hidden"]');
+        input.daterangepicker({
+          drops: 'down',
+          opens: 'center',
+          autoUpdateInput: false,
+          timePicker: true,
+          timePicker24Hour: true,
+          ranges: {
+            '$escapetool.javascript($services.localization.render("daterange.today"))':
+              [moment().startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.yesterday"))':
+              [moment().subtract(1, 'days').startOf('day'), moment().subtract(1, 'days').endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.lastSevenDays"))':
+              [moment().subtract(6, 'days').startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.lastThirtyDays"))':
+              [moment().subtract(29, 'days').startOf('day'), moment().endOf('day')],
+            '$escapetool.javascript($services.localization.render("daterange.thisMonth"))':
+              [moment().startOf('month'), moment().endOf('month')],
+            '$escapetool.javascript($services.localization.render("daterange.lastMonth"))':
+              [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
+          },
+          locale: {
+            format: dateFormat,
+            cancelLabel: '$escapetool.javascript($services.localization.render("daterange.clear"))',
+            applyLabel: '$escapetool.javascript($services.localization.render("daterange.apply"))',
+            customRangeLabel: '$escapetool.javascript($services.localization.render("daterange.customRange"))',
+            fromLabel: '$escapetool.javascript($services.localization.render("daterange.from"))',
+            toLabel: '$escapetool.javascript($services.localization.render("daterange.to"))',
+          }
+        });
+
+        var updateInput = function(element, event, picker) {
+          if (event.type == 'cancel') {
+            input.val('');
+            hidden.val('');
+          } else {
+            if (picker.startDate.isSame(picker.endDate, 'day')) {
+              input.val(picker.startDate.format(dateFormat));
+            } else {
+              input.val(picker.startDate.format(dateFormat) + ' - ' + picker.endDate.format(dateFormat));
+            }
+            hidden.val(picker.startDate.format('x') + '-' + picker.endDate.format('x'));
+          }
+          if ('dispatchEvent' in element) {
```

Comment:
First, I think you need to add a comment to explain why you need to trigger the "change" event. I suppose it's because you want to update the live table rows. Now, if I look at https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetable.js#L999 I see that the livetable is using ``Event.observe`` to register its listener, so I still don't understand why calling ``element.fire('change')`` or ``Event.fire(element, 'change')`` doesn't call call this listener. If you don't want to debug then there's still a better way to trigger the refresh of the live table, see https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetable.js#L1008 (i.e. trigger the custom event). Since this is a custom event, unlike "change", you could even trigger it from jQuery, if you add ``xwiki-event-bridge`` as a dependency to your Require.js module. WDYT?